### PR TITLE
Revert "PRNG cleanup and tests for NoisyLTFArray"

### PR DESCRIPTION
**Pull request by MrM0nkey**
Reverts nils-wisiol/pypuf#24

Running the sim_learn gives me this error

```
python -m sim_learn "16" "2" "id" "xor" "12000" "1" "10" "1234" "1234" Learning 16-bit 2 XOR Arbiter PUF with 12000 CRPs and 1 restarts.

Using
  transformation:       <function LTFArray.transform_id at 0x7fcff2529ea0>
  combiner:             <function LTFArray.combiner_xor at 0x7fcff2529a60>
  instance random seed: 0x1234
  model random seed:    0x1234

----------- Choosing new instance. ---------
    1/    1         Traceback (most recent call last):
  File "/home/christoph/miniconda3/envs/pypuf/lib/python3.4/runpy.py", line 170, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/christoph/miniconda3/envs/pypuf/lib/python3.4/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/christoph/arbeit/pypuf/sim_learn.py", line 116, in <module>
    model = lr_learner.learn()
  File "/home/christoph/arbeit/pypuf/pypuf/learner/regression/logistic_regression.py", line 202, in learn
    weight_array=LTFArray.normal_weights(self.n, self.k, self.weights_mu, self.weights_sigma, self.weight_prng),
AttributeError: 'LogisticRegression' object has no attribute 'weight_prng'
```

Please fix this and add a simple test that runs sim_learn.py so this will not happen again.

### CODE REVIEW

File: test/test_logistic_regression.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
+import unittest
+from numpy.random import RandomState
+from pypuf.simulation.arbiter_based.ltfarray import LTFArray
+from pypuf.learner.regression.logistic_regression import LogisticRegression
+from pypuf.tools import TrainingSet
+
+
+class TestLogisticRegression(unittest.TestCase):
+    """
+        This module should test the logistic regression learner.
```
Comment (nils-wisiol) [inline test/test_logistic_regression.py]: 'should'? It either does or doesnt :)

File: test/test_logistic_regression.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    """
+    n = 16
+    k = 2
+    N = 12000
+    seed_model = 1234
+    seed_instance = 1234
+
+    def test_learn_xor(self):
+        """"
+            Stupid test which should gain code coverage
```
Comment (nils-wisiol) [inline test/test_logistic_regression.py]: 'should'

File: test/test_logistic_regression.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+            TestLogisticRegression.k,
+            transformation=LTFArray.transform_id,
+            combiner=LTFArray.combiner_xor,
+            weights_prng=model_prng,
+        )
+        lr_learner.learn()
+
+    def test_learn_ip_mod2(self):
+        """"
+            Stupid test which should gain code coverage
```
Comment (nils-wisiol) [inline test/test_logistic_regression.py]: 'should'

**Comment by nils-wisiol** [comment]
The first commit ("Added testcases..." 9169d699c8704b1be9b65a29ba75bb0a9fa410d0) doesn't introduce any changes?

For the last commit, I like the tests, but they don't increase coverage, as this was all already tested in sim_learn.py. I don't mind them though.

For all commits, please rework the commit messages to fit GitHub's standards (no more than 80 chars)

File: test/test_logistic_regression.py (hunk 3 above)
Comment (MrM0nkey) [inline test/test_logistic_regression.py]: Done  with https://github.com/nils-wisiol/pypuf/pull/28/commits/df83fcc7968b9ea161fbcf8dd6658068db0637db

File: test/test_logistic_regression.py (hunk 2 above)
Comment (MrM0nkey) [inline test/test_logistic_regression.py]: Done with https://github.com/nils-wisiol/pypuf/pull/28/commits/df83fcc7968b9ea161fbcf8dd6658068db0637db

File: test/test_logistic_regression.py (hunk 1 above)
Comment (MrM0nkey) [inline test/test_logistic_regression.py]: Done with https://github.com/nils-wisiol/pypuf/pull/28/commits/df83fcc7968b9ea161fbcf8dd6658068db0637db

**Comment by nils-wisiol** [comment]
lgtm

**Pull request merged**